### PR TITLE
Try to obtain --ap-mso from .netrc

### PR DESCRIPTION
**Pull request by brbsix**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

As a fallback in the event that no `--ap-mso` is explicitly supplied, iterate over all available credentials (params and .netrc) for the first supported one.

### CODE REVIEW

**Comment by dstftw** [comment]
Did you even bother to run your code?
```
PS C:\dev\youtube-dl\master> py -3.6 .\youtube_dl\__main__.py http://watchdisneychannel.go.com/liv-and-maddie/video/vdka3745751/04/15-endarooney
[Go] 15-endarooney: Downloading webpage
[Go] 3745751: Downloading JSON metadata
Traceback (most recent call last):
  File ".\youtube_dl\__main__.py", line 19, in <module>
    youtube_dl.main()
  File "C:\dev\youtube-dl\master\youtube_dl\__init__.py", line 464, in main
    _real_main(argv)
  File "C:\dev\youtube-dl\master\youtube_dl\__init__.py", line 454, in _real_main
    retcode = ydl.download(all_urls)
  File "C:\dev\youtube-dl\master\youtube_dl\YoutubeDL.py", line 1890, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "C:\dev\youtube-dl\master\youtube_dl\YoutubeDL.py", line 761, in extract_info
    ie_result = ie.extract(url)
  File "C:\dev\youtube-dl\master\youtube_dl\extractor\common.py", line 429, in extract
    ie_result = self._real_extract(url)
  File "C:\dev\youtube-dl\master\youtube_dl\extractor\go.py", line 98, in _real_extract
    url, video_id, requestor_id, resource)
  File "C:\dev\youtube-dl\master\youtube_dl\extractor\adobepass.py", line 1371, in _extract_mvpd_auth
    username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
  File "C:\dev\youtube-dl\master\youtube_dl\extractor\common.py", line 832, in _get_login_info
    username, password = self._get_netrc_login_info(netrc_machine)
  File "C:\dev\youtube-dl\master\youtube_dl\extractor\common.py", line 796, in _get_netrc_login_info
    netrc_machine = netrc_machine or self._NETRC_MACHINE
AttributeError: 'GoIE' object has no attribute '_NETRC_MACHINE'
```

File: youtube_dl/extractor/adobepass.py [hunk 1]
```diff
@@ -1362,11 +1362,17 @@ def raise_mvpd_required():
...
-                username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
-                if not username or not password:
-                    raise_mvpd_required()
+                if mso_id:
+                    username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
+                    if not username or not password:
+                        raise_mvpd_required()
+                else:
+                    for mso in MSO_INFO:
+                        username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
```
Comment (dstftw) [inline youtube_dl/extractor/adobepass.py]: `mso_id` is always `None` at this point.

File: youtube_dl/extractor/adobepass.py [hunk 2]
```diff
@@ -1362,11 +1362,17 @@ def raise_mvpd_required():
...
-                    raise_mvpd_required()
-                username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
-                if not username or not password:
-                    raise_mvpd_required()
+                if mso_id:
+                    username, password = self._get_login_info('ap_username', 'ap_password', mso_id)
+                    if not username or not password:
+                        raise_mvpd_required()
+                else:
+                    for mso in MSO_INFO:
```
Comment (dstftw) [inline youtube_dl/extractor/adobepass.py]: This introduces an ambiguity in case of several mso available.

**Comment by brbsix** [comment]
Sorry, I probably should have proposed this in an issue rather than a half-baked PR. I wasn't sure whether this behavior would even be desirable as a new default (e.g. as you mentioned in the case that there are multiple MSOs in the .netrc do we want to go ahead and try the first one or fail with an error?), just that it was useful to me. For what it's worth, I've rewritten the branch with working code. https://github.com/brbsix/youtube-dl/commit/b27c0b433cc79639866c6b8ae245ac15bc31bfb2

**Comment by raleeper** [comment]
@brbsix FWIW, I just put my default MSO setting into the user config file. Works even if the current site doesn't require MSO authorization.

**Pull request closed without merging**